layouts/shortcodes/api-reference-link.html

```diff
@@ -0,0 +1,41 @@
+<!-- Check if 'api-metadata' exists in the front-matter of the file -->
+{{- if .Page.Params.api_metadata -}}
+<!-- Extract API metadata -->
+{{- $apiVersion := .Page.Params.api_metadata.apiVersion -}}
+{{- $kind := .Page.Params.api_metadata.kind -}}
+{{- $type := .Page.Params.api_metadata.type -}}
+{{- $version := $apiVersion -}}
+
+<!-- Define variable to store API reference links -->
+{{- $apiReferenceLinks := "" -}}
+{{- $apiReferenceText := T "api_reference_title" -}}
+
+<!-- Get all files under the specified directory -->
+{{- $apiRefDir := printf "docs/reference/kubernetes-api/%s-resources" $type -}}
+{{- $apiReferenceFiles := site.GetPage $apiRefDir -}}
+
+<!-- Loop through API reference files -->
+{{- range $apiRefFile := $apiReferenceFiles.RegularPages -}}
+  {{- if and (ne .Section "") (in .File.Dir $apiRefDir) -}}
+    <!-- Check if the file's metadata matches -->
+    {{- with .Params.api_metadata -}}
+      {{- if and (eq .kind $kind) (eq .apiVersion $version) -}}
+        <!-- If the file's metadata matches, add link to variable -->
+        {{- $link := printf "<a id='api-reference-page-link' href='%s'>%s %s</a>" $apiRefFile.Permalink $kind $apiReferenceText -}}
+        {{- $apiReferenceLinks = printf "%s %s" $apiReferenceLinks $link -}}
+      {{- end -}}
+    {{- end -}}
+  {{- end -}}
+{{- end -}}
+
+<!-- Display API reference heading and link if there is any -->
+{{- if ne $apiReferenceLinks "" -}}
+  <h2 id="api-reference">{{- $apiReferenceText -}}</h2>
+  <p>
+  {{- $conceptPageApiRefSection := T "api_reference_text" -}}
+  {{- $conceptPageApiRefSection = replace $conceptPageApiRefSection "<resource-api-ref>" $apiReferenceLinks -}}
```

Comment:
I'd use a printf here; that will fail at render time if there is no placeholder in the string for a localization.

Reply:
I've revised the text now. Let me know if it needs further enhancement.